view/frontend/web/js/view/payment/method-renderer/gr4vy-method.js

```diff
@@ -174,7 +134,42 @@ define(
                     });
             },
             /**
-             * Place order in Magento before gr4vy transaction
+             * Process Gr4vy Transaction and Place an Order in Magento
+             */
+            processGr4vyOrder: function (transaction) {
+                var self = this;
+                var serviceUrl = urlBuilder.createUrl('/gr4vy-payment/process-gr4vy-transaction', {});
+                var payload = {
+                    cartId: quote.getQuoteId(),
+                    transactionId: transaction.id,
+                    paymentMethod: this.getPaymentMethodData(),
+                    methodData: this.getGr4vyPaymentMethodData(),
+                    serviceData: this.getGr4vyPaymentServiceData(),
+                    transactionData: this.getGr4vyTransactionData()
```

Comment:
Do we need to set these properties and call the functions with empty values even when these are populated by the server function?

Reply:
👍